lib/galaxy/managers/users.py

```diff
@@ -840,13 +837,20 @@ def _add_parsers(self):
         self.fn_filter_parsers.update({})
 
 
-def get_user_by_username(session, user_class, username):
-    """
-    Get a user from the database by username.
-    (We pass the session and the user_class to accommodate usage from the tool_shed app.)
-    """
-    try:
-        stmt = select(user_class).filter(user_class.username == username)
-        return session.execute(stmt).scalar_one()
-    except Exception:
-        return None
+def get_users_by_ids(session: Session, user_ids):
+    stmt = select(User).where(User.id.in_(user_ids))
+    return session.scalars(stmt).all()
+
+
+# The get_user_by_email and get_user_by_username functions may be called from
```

Comment:
How about a different function for the tool shed, then you also don't need to use `trans.model.User` ?

Reply:
I tried - that was my first solution. It didn't work: tool shed is tightly coupled: this function is called indirectly as part of the authentication process (as far as I remember) by both galaxy and tool shed. Changing this behavior requires refactoring tool shed code which doesn't have sufficient test coverage, and I was reluctant to touch it as part of this PR.